### PR TITLE
Add Publisher serializer

### PR DESCRIPTION
**Pull request by goosemania**
re #2298
https://pulp.plan.io/issues/2298

### CODE REVIEW

**Comment by goosemania** [comment]
This is done similar to Importer Serializer #2806. This PR should wait for the resolution of the nested routing stuff, till then -  work-in-progress.

File: app/pulp/app/viewsets/repository.py [hunk 1]
```diff
@@ -64,3 +75,14 @@ class ImporterViewSet(NamedModelViewSet):
...
     serializer_class = ImporterSerializer
     queryset = Importer.objects.all()
+
+
+class PublisherViewSet(NamedModelViewSet):
+    # Indicates that publisher urls should be routed through their associated repository.
+    nested_parent = RepositoryViewSet
+    # Name the parameter to generate the link. This is important because `name` in this url refers
+    # to `Publisher.name`, not `Repository.name`
+    nested_parent_lookup_name = 'repo_name'
```
Comment (asmacdo) [inline app/pulp/app/viewsets/repository.py]: The previous 5 lines all need to go.

**Comment by goosemania** [comment]
@asmacdo, I guess I have to add a repository field as well because we are no longer counting on the nested thing. Is it correct or I miss something?

**Comment by goosemania** [comment]
@asmacdo, I did all the changes we discussed, it is ready for re-review I think. Thanks!

**Comment by asmacdo** [comment]
<LGTM when done>

Also, I really like the link to the issue in the commit, going to start doing that.

**Pull request merged**